File: app/src/main/java/dev/ragnarok/fenrir/api/services/IUploadService.java
package dev.ragnarok.fenrir.api.services;

import dev.ragnarok.fenrir.api.model.response.BaseResponse;
import dev.ragnarok.fenrir.api.model.upload.UploadAudioDto;
import dev.ragnarok.fenrir.api.model.upload.UploadChatPhotoDto;
import dev.ragnarok.fenrir.api.model.upload.UploadDocDto;
import dev.ragnarok.fenrir.api.model.upload.UploadOwnerPhotoDto;
import dev.ragnarok.fenrir.api.model.upload.UploadPhotoToAlbumDto;
import dev.ragnarok.fenrir.api.model.upload.UploadPhotoToMessageDto;
import dev.ragnarok.fenrir.api.model.upload.UploadPhotoToWallDto;
import dev.ragnarok.fenrir.api.model.upload.UploadStoryDto;
import dev.ragnarok.fenrir.api.model.upload.UploadVideoDto;
import io.reactivex.rxjava3.core.Single;
import okhttp3.MultipartBody;
import retrofit2.http.Multipart;
import retrofit2.http.POST;
import retrofit2.http.Part;
import retrofit2.http.Url;

public interface IUploadService {

    @Multipart
    @POST
    Single<UploadDocDto> uploadDocumentRx(@Url String server, @Part MultipartBody.Part file);

    @Multipart
    @POST
    Single<UploadAudioDto> uploadAudioRx(@Url String server, @Part MultipartBody.Part file);

    @Multipart
    @POST
    Single<BaseResponse<Integer>> remotePlayAudioRx(@Url String server, @Part MultipartBody.Part file);

    @Multipart
    @POST
    Single<BaseResponse<UploadStoryDto>> uploadStoryRx(@Url String server, @Part MultipartBody.Part file);

    @Multipart
    @POST
    Single<UploadVideoDto> uploadVideoRx(@Url String server, @Part MultipartBody.Part file);

    @Multipart
    @POST
    Single<UploadOwnerPhotoDto> uploadOwnerPhotoRx(@Url String server, @Part MultipartBody.Part photo);

    @Multipart
    @POST
    Single<UploadChatPhotoDto> uploadChatPhotoRx(@Url String server, @Part MultipartBody.Part photo);

    @Multipart
    @POST
    Single<UploadPhotoToWallDto> uploadPhotoToWallRx(@Url String server, @Part MultipartBody.Part photo);

    @Multipart
    @POST
    Single<UploadPhotoToMessageDto> uploadPhotoToMessageRx(@Url String server, @Part MultipartBody.Part photo);

    @Multipart
    @POST
    Single<UploadPhotoToAlbumDto> uploadPhotoToAlbumRx(@Url String server, @Part MultipartBody.Part file1);
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/api/services/IUploadService.kt
package dev.ragnarok.fenrir.api.services

import dev.ragnarok.fenrir.api.model.VKApiJsonString
import dev.ragnarok.fenrir.api.model.response.BaseResponse
import dev.ragnarok.fenrir.api.model.response.CustomResponse
import dev.ragnarok.fenrir.api.model.upload.UploadAudioDto
import dev.ragnarok.fenrir.api.model.upload.UploadChatPhotoDto
import dev.ragnarok.fenrir.api.model.upload.UploadDocDto
import dev.ragnarok.fenrir.api.model.upload.UploadOwnerPhotoDto
import dev.ragnarok.fenrir.api.model.upload.UploadPhotoToAlbumDto
import dev.ragnarok.fenrir.api.model.upload.UploadPhotoToMessageDto
import dev.ragnarok.fenrir.api.model.upload.UploadPhotoToWallDto
import dev.ragnarok.fenrir.api.model.upload.UploadStoryDto
import dev.ragnarok.fenrir.api.model.upload.UploadVideoDto
import dev.ragnarok.fenrir.api.rest.IServiceRest
import kotlinx.coroutines.flow.Flow
import okhttp3.MultipartBody

class IUploadService : IServiceRest() {
    fun uploadDocumentRx(server: String, file: MultipartBody.Part): Flow<UploadDocDto> {
        return rest.doMultipartFormFullUrl(server, file, UploadDocDto.serializer())
    }

    fun uploadAudioRx(server: String, file: MultipartBody.Part): Flow<UploadAudioDto> {
        return rest.doMultipartFormFullUrl(server, file, UploadAudioDto.serializer())
    }

    fun remotePlayAudioRx(
        server: String,
        file: MultipartBody.Part
    ): Flow<BaseResponse<Int>> {
        return rest.doMultipartFormFullUrl(server, file, baseInt)
    }

    fun uploadStoryPhotoRx(
        server: String,
        file: MultipartBody.Part
    ): Flow<VKApiJsonString> {
        return rest.doMultipartFormFullUrl(
            server,
            file,
            VKApiJsonString.serializer()
        )
    }

    fun uploadStoryVideoRx(
        server: String,
        file: MultipartBody.Part
    ): Flow<CustomResponse<UploadStoryDto>> {
        return rest.doMultipartFormFullUrl(
            server,
            file,
            CustomResponse.serializer(UploadStoryDto.serializer())
        )
    }

    fun uploadVideoRx(server: String, file: MultipartBody.Part): Flow<UploadVideoDto> {
        return rest.doMultipartFormFullUrl(server, file, UploadVideoDto.serializer())
    }

    fun uploadOwnerPhotoRx(
        server: String,
        photo: MultipartBody.Part
    ): Flow<UploadOwnerPhotoDto> {
        return rest.doMultipartFormFullUrl(server, photo, UploadOwnerPhotoDto.serializer())
    }

    fun uploadChatPhotoRx(
        server: String,
        photo: MultipartBody.Part
    ): Flow<UploadChatPhotoDto> {
        return rest.doMultipartFormFullUrl(server, photo, UploadChatPhotoDto.serializer())
    }

    fun uploadPhotoToWallRx(
        server: String,
        photo: MultipartBody.Part
    ): Flow<UploadPhotoToWallDto> {
        return rest.doMultipartFormFullUrl(server, photo, UploadPhotoToWallDto.serializer())
    }

    fun uploadPhotoToMessageRx(
        server: String,
        photo: MultipartBody.Part
    ): Flow<UploadPhotoToMessageDto> {
        return rest.doMultipartFormFullUrl(server, photo, UploadPhotoToMessageDto.serializer())
    }

    fun uploadPhotoToAlbumRx(
        server: String,
        file1: MultipartBody.Part
    ): Flow<UploadPhotoToAlbumDto> {
        return rest.doMultipartFormFullUrl(server, file1, UploadPhotoToAlbumDto.serializer())
    }
}
